### PR TITLE
Merge PR's branch with master before testing

### LINKED ISSUE #407: test-origin-image: merge PR with master?
I noticed that in [`lib/vagrant-openshift/command/test_origin_image.rb`](https://github.com/openshift/vagrant-openshift/blob/abbcc3ba653b11690ed200262c7e000dd7bc6351/lib/vagrant-openshift/command/test_origin_image.rb#L120-L140) we just checkout a PR and run tests without attempting to merge it with master:

```bash
# Fetch refs from Github pull requests
git fetch --quiet --tags --progress #{source} +refs/pull/*:refs/remotes/origin/pr/*

# switch to the desired ref
git checkout #{ref}

# ...

make test TARGET=rhel7
# ...
```

I think we should be running tests against the result of merging the PR with master, so that we don't rely solely on the state of the tree in the proposed branch.

@bparees @mfojtik @danmcp thoughts?

### PR DESCRIPTION
**Pull request by rhcarvalho**
Fixes #407.

### CODE REVIEW

**Comment by rhcarvalho** [comment]
@bparees @danmcp PTAL

Sorry this is not top priority, but was a too low hanging fruit that will make me trust better the test results for image change PRs.

I wonder what we do for testing origin? It wasn't very straight forward for me to browse the code and figure out exactly what's being done, I could find no clear code indicating a git merge happening.

File: lib/vagrant-openshift/command/test_origin_image.rb [hunk 1]
```diff
@@ -117,14 +118,14 @@ def execute
 cd $temp_dir
 
 # clone the image repo
-git clone #{source}
-cd #{source_dir}
+git clone "#{source}"
```
Comment (bparees) [inline lib/vagrant-openshift/command/test_origin_image.rb]: is this the upstream (openshift) repo?

Comment (rhcarvalho) [inline lib/vagrant-openshift/command/test_origin_image.rb]: No, this is one of the image repos:

https://github.com/openshift/vagrant-openshift/pull/413/files#diff-b5524b94a3165b89ca0779574f0c98c9R78

Typical values will be 

- https://github.com/openshift/mongodb
- https://github.com/openshift/mysql
- etc.

And FWIW `ref` is the pull request's number.

Comment (rhcarvalho) [inline lib/vagrant-openshift/command/test_origin_image.rb]: > is this the upstream (openshift) repo?

Oh well, on first thought I was thinking "origin", sorry.

The answer is **yes**, that's the upstream repo. The pull request ref is in the same repo.

Comment (bparees) [inline lib/vagrant-openshift/command/test_origin_image.rb]: ok great, thanks.

**Comment by bparees** [comment]
lgtm.

**Pull request merged**